protocol/invocation/rpcinvocation.go

```diff
@@ -213,6 +221,13 @@ func WithParameterTypes(parameterTypes []reflect.Type) option {
 	}
 }
 
+// WithParameterTypeNames creates option with @parameterTypeNames.
+func WithParameterTypeNames(parameterTypeNames []string) option {
+	return func(invo *RPCInvocation) {
+		invo.parameterTypeNames = parameterTypeNames
```

Comment:
IMO, u'd better copy this parameterTypeNames.

Reply:
done